### DISCUSSION TITLE
Timing out when adding an Intergroup's source

### DISCUSSION THREAD

**Discussion by fritzisa** [code4recovery/12-step-meeting-list#591] [General]
I ran into an error message when adding an Intergroup's source to our Area's website.
This is the first time we've had a problem.

The source link they provided:
[(https://york-pa-aa.org/wp-admin/admin-ajax.php?action=meetings&key=0571c77c2ddabe26ac1f98ce1a5d15bd)]

Our website/meetings:
[(https://area59aa.org/meetings/]

Any assistance would be greatly appreciated!

<img width="1259" alt="Screen Shot 2021-12-07 at 2 09 41 PM" src="https://user-images.githubusercontent.com/95769327/145252519-9b33f8fe-6016-43d0-b785-e39bff91b543.png">

**Comment by joshreisner** [comment]
i'm going to move this to the 12-step-meeting-list repo, but my strong suspicion is they have some kind of firewall that's blocking the request. i see that browser requests work, but if i ping that server from the command line i also get a request timeout

**Comment by marcinyorkpa** [comment]
@joshreisner I'm running the problematic website that @fritzisa is talking about. When you say you ping the server you get a request timeout, do you mean literally `ping` or do you mean something like `curl "https://york-pa-aa.org/wp-admin/admin-ajax.php?action=meetings&key=0571c77c2ddabe26ac1f98ce1a5d15bd"` ?

I ask because ping definitely won't work, but all `curl`s I've done from my own servers on different networks work fine for that URL, so I haven't actually been able to replicate the timeout behavior, and I don't see any evidence of it in our web logs.

I did find a firewall of sorts on our shared host that I'll try to dig into, but I'd expect that the Meeting Guide app's nightly fetch of our meeting data would also be failing... but it's been working fine for years. So, basically, I'm puzzled.

Thanks again for any help!

**Comment by brianw-area24** [comment]
I finally got around to giving this a try. I was able to successfully run the curl command from terminal. Therefore, I put the feed URL into my test WordPress site, and was able to successfully pull the meetings into my test site (108 meetings).

I'm not sure if @marcinyorkpa changed anything or not. Curious if @fritzisa is still having problems with this...

**Comment by joshreisner** [comment]
@marcinyorkpa sorry i must have been confused yesterday about ping. i do get a request timeout when i `ping york-pa-aa.org` but i'm sure that's not related. i just tested and importing your feed works fine on my demo site.

i was thinking specifically of meeting guide; firewalls on aa websites are so common, the fix is often to ask a site to create an exception in their firewall for the importing website. but not in this case. not sure what the issue is here but im guessing it must be on the area 59 setup somehow. perhaps there is a firewall for outbound connections? is that a thing?

**Comment by fritzisa** [comment]
I just tried again and I still get this message when I try to load their data feed:

Import & Settings

Invalid response,

WP_Error Object
(
    [errors] => Array
        (
            [http_request_failed] => Array
                (
                    [0] => cURL error 28: Connection timeout after 10000 ms
                )

        )

    [error_data] => Array
        (
        )

    [additional_data:protected] => Array
        (
        )

)
.

So it's something on my end.

FWIW: I have been adding data feeds recently without any problems.

Any suggestions would be really helpful.

Thanks,
Scott

**Comment by joshreisner** [comment]
@fritzisa i would first look at any security-related plugins you might have, and try temporarily disabling them and retrying the import. if that doesn't help, i would reach out to your hosting company and asking if there's a reason your site can't make an outbound curl request

**Comment by fritzisa** [comment]
Thank you @joshreisner 
That's what I'll do.

**Comment by fritzisa** [comment]
I disabled our Securi firewall and tried again and got the same error message. I worked with the webhost (GoDaddy) to troubleshoot every possible cause and they come to the conclusion it is the script. I am at a total loss as this is the only data feed out of (12) that I've had any problem with. I'll reconnect with Marc at York Intergroup to make sure all the steps provided have been taken.

**Comment by joshreisner** [comment]
Wow, very strange. Sorry, I didn't realize you had other feeds active.

I genuinely have no idea why this is happening. I guess the next thing I would try to confirm, which is probably hard, is if there is some kind of routing problem between your server and york-pa-aa.org. I doubt GoDaddy will be of much help with that.

I'm not sure what we can do on our end, since it's working on our demo site. For example, increasing the timeout.

Are you allowed to SSH into your server?

Anyone on the C4R side have other ideas of what to try?

**Comment by fritzisa** [comment]
Marc from York Intergroup and I are going to connect on Zoom and do some screen sharing. We'll let you know what we come up with. Thanks for all the great suggestions ... "more will be revealed" I am sure ;)

**Reply by tech2serve** [reply]
Any luck?

I too am baffled, but sometimes I’ve found in these weird situations that getting both sides to clear server-side caches can clear up weird and inconsistent behaviors. Yeah, I know it is a long shot but….

Are either of you watching the dev/debugging console when you try the import? Any other errors/warnings/notifications?

**Comment by tech2serve** [comment]
I’m going to transfer this to the TSML Discussions page (to help keep clean lanes between support and bugs/features to be worked).

If this does turn out to be a bug, we’ll open a new issue focused on that.

**Comment by marcinyorkpa** [comment]
@tech2serve @fritzisa Nope, no luck. We confirmed that it's not a problem with the actual php endpoint or "key".. the request is simply never making it to our york-pa-aa.org site at all, but we have no idea why. It sure feels firewall-ish but neither of us had any luck with disabling firewalls we had access to or plugins. After that it always resulted in the same behavior. I tried working with our host to see if they could help, but it was fruitless, as  they basically said "we don't have firewalls". 

So the Area 59 site can reach out successfully to a bunch of other TSML exports, but not ours.

And the Meeting Guide app can successfully reach into our TSML export, but Area 59's can't.

It's mystifying. I googled for PHP curl error 28, and most of what I found was "firewall / network / etc" related.

Scott wasn't familiar with how to turn on wordpress debugging -- and honestly I don't know how to do it except modifying wp-config.php and redeploying it and I don't think Scott was comfortable doing that on his site (understandably). If you know of a better way to see additional error / debug info beyond what TSML puts into the import box, please let us know.

For now, I'm just going to export a CSV file for Area 59 to import every month or so.

Thanks for all the help.

**Comment by sriphoenix** [comment]
I was able to fix this problem on my end staging.aaphoenix.org.  I use the plugin- IThemes Security Pro.  I needed to change the setting under Advanced, Wordpress Tweaks, to Enable XML-RPC. It was set to Disable XML-RPC.   Hope this helps any others that get this problem.
